File: trunk/tests/CoreTests/Exceptions/Class1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SharpKit.JavaScript;

namespace CoreTests.Exceptions
{
    [JsType(JsMode.Prototype)]
    class Class1
    {
        void Foo()
        {
            try
            {
            }
            catch
            {
                throw;
            }
            try
            {
            }
            catch(Exception)
            {
                throw;
            }

            try
            {
            }
            catch (Exception e)
            {
                throw;
            }
            try
            {
            }
            catch (Exception e)
            {
                throw e;
            }
        }

    }
}

File: trunk/tests/CoreTests/Exceptions/Class1.js
;
if (typeof(CoreTests) == "undefined")
    var CoreTests = {};
if (typeof(CoreTests.Exceptions) == "undefined")
    CoreTests.Exceptions = {};
CoreTests.Exceptions.Class1 = function (){
};
CoreTests.Exceptions.Class1.prototype.Foo = function (){
    try{
    }
    catch($$e1){
        throw $CreateException($$e1, new Error());
    }
    try{
    }
    catch($$e2){
        throw $CreateException($$e2, new Error());
    }
    try{
    }
    catch(e){
        throw $CreateException(e, new Error());
    }
    try{
    }
    catch(e){
        throw $CreateException(e, new Error());
    }
};

